File: src/rank.rs
use std::fmt::Display;

#[derive(PartialEq, PartialOrd)]
pub enum Rank {
    Unrated = 0,
    Gold = 1,
    Emerald = 2,
    Diamond = 3,
}

impl From<u16> for Rank {
    fn from(value: u16) -> Self {
        if value >= 1500 {
            Rank::Diamond
        } else if value >= 1300 {
            Rank::Emerald
        } else if value >= 1100 {
            Rank::Gold
        } else {
            Rank::Unrated
        }
    }
}

impl Display for Rank {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            Rank::Unrated => write!(f, "Unrated"),
            Rank::Gold => write!(f, "Gold"),
            Rank::Emerald => write!(f, "Emerald"),
            Rank::Diamond => write!(f, "Diamond"),
        }
    }
}

File: player/rank.py
from enum import Enum


class Rank(Enum):
    Diamond = 1
    Emerald = 2
    Gold = 3
    Unrated = 4

    def __repr__(self):
        return f"{self._name_}"

    def __ge__(self, other):
        if self.__class__ is other.__class__:
            return self.value >= other.value
        return NotImplemented

    def __gt__(self, other):
        if self.__class__ is other.__class__:
            return self.value > other.value
        return NotImplemented

    def __le__(self, other):
        if self.__class__ is other.__class__:
            return self.value <= other.value
        return NotImplemented

    def __lt__(self, other):
        if self.__class__ is other.__class__:
            return self.value < other.value
        return NotImplemented


def determine_rank(points: int) -> Rank:
    if points >= 1500:
        return Rank.Diamond
    elif points >= 1300:
        return Rank.Emerald
    elif points >= 1100:
        return Rank.Gold
    else:
        return Rank.Unrated
